src/main/java/ch/wisv/areafiftylan/users/controller/UserRestController.java

```diff
@@ -164,4 +165,9 @@ public Boolean checkUsernameExists(@RequestParam String username) {
     public ResponseEntity<?> handleDataIntegrityViolationException(DataIntegrityViolationException ex) {
         return createResponseEntity(HttpStatus.CONFLICT, "Username or Email already taken!");
     }
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    public ResponseEntity<?> handleConstraintViolationException(ConstraintViolationException ex) {
+        return createResponseEntity(HttpStatus.BAD_REQUEST, "Username should be a valid Email!");
```

Comment:
Is this the only possible cause for this Exception? ConstraintViolationException sounds like it's more generic then just an invalid email field.